modules/home-manager.nix

```
@@ -120,6 +122,10 @@ in
 
       environment.packages = mkIf cfg.useUserPackages cfg.config.home.packages;
 
+      environment.etc = if (! cfg.useUserPackages) then {} else {
+        "profiles/per-user/${config.user.userName}".source =
+          builtins.toPath "${config.user.home}/.nix-profile";
+      };
```

```suggestion
      environment.etc = mkIf cfg.useUserPackages {
        "profiles/per-user/${config.user.userName}".source =
          "${config.user.home}/.nix-profile";
      };
```

And maybe add a small comment, why this workaround is needed. But seems legit to me as a working fix.

---

Added a comment, changed to mkIf. Makes me worried about how many times I should've used mkIf in the past and hasn't.